tools/update_runnablejobs.py

```diff
@@ -0,0 +1,110 @@
+
+from redo import retry
+
+headers = {
+    'Accept': 'application/json',
+    'User-Agent': 'ouija',
+}
+
+TREEHERDER_HOST = "https://treeherder.mozilla.org/api/project/{0}/" \
+                  "runnable_jobs/?decisionTaskID={1}&format=json"
+ROOT_DIR = os.getcwd()
+
+
+# TODO we need add test for this function to make sure it works under any situation
+def update_runnableapi():
+    """Use it to update runnablejobs.json file."""
+    url = "https://index.taskcluster.net/v1/task/gecko.v2.%s.latest.firefox.decision/"
+    latest_task = retry(requests.get, args=(url % "mozilla-inbound", ),
+                        kwargs={'headers': {'accept-encoding': 'json'}, 'verify': True}).json()
+    task_id = latest_task['taskId']
+
+    # The format of expires is like 2017-07-04T22:13:23.248Z and we only want 2017-07-04 part
+    expires = latest_task['expires'].split('T')[0]
+    time_tuple = datetime.datetime.strptime(expires, "%Y-%m-%d").timetuple()
+    new_timestamp = time.mktime(time_tuple)
+    path = ROOT_DIR + '/runnablejobs.json'
+    # we do nothing if the timestamp of runablejobs.json is equal with the latest task
+    # otherwise we download and update it
+    if os.path.isfile(path):
+        with open(path, 'r+') as data:
+            # read the timesstamp of this task from json file
+            oldtime = json.loads(data.read())['meta']['timetamp']
+        if oldtime == new_timestamp:
+            print "The runnable json file is latest already."
+            return
+        else:
+            print "It's going to update your runnable jobs data."
+            data = query_the_runnablejobs(new_timestamp, task_id)
+            add_new_jobs_into_pressed(data=data)
+            with open(ROOT_DIR + '/runnablejobs.json', 'w') as f:
+                json.dump(data, f)
+    else:
+        print "It's going to help you download the runnable jobs file."
+        data = query_the_runnablejobs(new_timestamp, task_id)
+        if data:
+            with open(ROOT_DIR + '/runnablejobs.json', 'w') as f:
+                json.dump(data, f)
+
+
+def query_the_runnablejobs(new_timestamp, task_id=None):
+    if task_id:
+        url = TREEHERDER_HOST.format('mozilla-inbound', task_id)
+        data = retry(requests.get, args=(url, ), kwargs={'headers': headers}).json()
+        if len(data['results']) > 0:
+            data['meta'].update({'timetamp': new_timestamp})
+            return data
+
+
+def add_new_jobs_into_pressed(new_data=None):
+    if new_data:
+        with open(ROOT_DIR + '/runnablejobs.json', 'w') as f:
+            old_data = json.loads(f.read())['jobtypes']
+        # If a job in recent runnablejobs and can't been found in the old runnablejobs,
+        # that job been determined as a new job and pick it up.
+        new_jobs = [new_job['ref_data_name'] for
+                    new_job in new_data if new_job in old_data]
+        if len(new_jobs) > 0:
+            # This part of code is actually duplicate with jobtypes.
```

Comment:
This part of code is duplicated, but I haven't found a better way to do this(get testname and other information we want to to pressed into the preseed.json as a new job.